cli/src/fty-asset-activator-cli.cpp

```diff
@@ -0,0 +1,75 @@
+#include <libfty_asset_activator.h>
+#include <fty/convert.h>
+#include <fty_common_mlm.h>
+
+#include <cstring>
+#include <string>
+#include <iostream>
+
+using namespace fty;
+
+void help()
+{
+    std::cout << "Usage: ./fty_asset_activator_cli <command> <asset iname>" << std::endl;
+    std::cout << "Available commands:" << std::endl;
+    std::cout << "\t- activate" << std::endl;
+    std::cout << "\t- deactivate" << std::endl;
+    std::cout << "\t- isactive" << std::endl;
+    std::cout << "\t- isactivable" << std::endl;
+}
+
+int getOption(const std::string& op)
+{
+    if(op == "activate") {
+        return 1;
+    } else if(op == "deactivate") {
+        return 2;
+    } else if(op == "isactive") {
+        return 3;
+    } else if(op == "isactivable") {
+        return 4;
+    }
+
+    return 0;
+}
+
+int main(int argc, char** argv)
+{
+    if(argc < 3) {
+        std::cout << "Invalid number of arguments" << std::endl;
+        help();
+
+        return 1;
+    }
+
+    std::string op(argv[1]);
+    std::string iname(argv[2]);
+
+    mlm::MlmSyncClient  client("asset-agent", "etn-licensing-credits");
+    AssetActivator activator(client);
+ 
+    bool ret;
+
+    switch(getOption(op)) {
```

Comment:
Just for fun... look into https://github.com/42ity/fty-utils/blob/master/fty/command-line.h
Example: https://github.com/42ity/fty-discovery-ng/blob/de145e0009d7c4f3dda6094d95edf68a0f292379/server/src/main.cpp#L15

Reply:
<3 I will need to implement a CLI for SRR, that will be really handy, thanks!